deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -694,3 +536,274 @@ def merge_input_provider(index, item):
         round_completion_file_s3_url,
         is_inplace_compacted,
     )
+
+
+def _process_merge_results(
+    params: CompactPartitionParams, mat_results, merge_results, compaction_audit
+):
+    for merge_result in merge_results:
+        mat_results.extend(merge_result.materialize_results)
+
+    mat_results: List[MaterializeResult] = sorted(
+        mat_results, key=lambda m: m.task_index
+    )
+
+    hb_id_to_entry_indices_range = {}
+    file_index = 0
+    previous_task_index = -1
+
+    for mat_result in mat_results:
+        assert (
+            mat_result.pyarrow_write_result.files >= 1
+        ), "Atleast one file must be materialized"
+        assert (
+            mat_result.task_index != previous_task_index
+        ), f"Multiple materialize results found for a hash bucket: {mat_result.task_index}"
+
+        hb_id_to_entry_indices_range[str(mat_result.task_index)] = (
+            file_index,
+            file_index + mat_result.pyarrow_write_result.files,
+        )
+
+        file_index += mat_result.pyarrow_write_result.files
+        previous_task_index = mat_result.task_index
+
+    s3_utils.upload(
+        compaction_audit.audit_url,
+        str(json.dumps(compaction_audit)),
+        **params.s3_client_kwargs,
+    )
+
+    deltas = [m.delta for m in mat_results]
+
+    # Note: An appropriate last stream position must be set
+    # to avoid correctness issue.
+    merged_delta: Delta = Delta.merge_deltas(
+        deltas,
+        stream_position=params.last_stream_position_to_compact,
+    )
+
+    return merged_delta, hb_id_to_entry_indices_range
+
+
+def _merge(
+    params: CompactPartitionParams,
+    task_resource_options_provider,
+    merge_resource_options_provider,
+    all_hash_group_idx_to_size_bytes,
+    all_hash_group_idx_to_num_rows,
+    round_completion_info,
+    previous_compacted_delta_manifest,
+    all_hash_group_idx_to_obj_id,
+    task_max_parallelism,
+    compacted_partition,
+    delete_strategy,
+    delete_file_envelopes,
```

Comment:
As a rule, avoid redundant arguments in every new method you create. Better to create a separate class that models input whose each arg brings in new information to work with.

Reply:
After going through each helper function call and the params present in CompactPartitionParams, it looks like task_max_parallelism was the redundant argument. This arg gets modified within _execute_compaction as shown below before going into my helper functions. Is there any danger in using the unmodified one present in params instead? Let me know if this question makes sense. 

`
task_max_parallelism: int = params.task_max_parallelism

    if params.pg_config:
        logger.info(
            "pg_config specified. Tasks will be scheduled in a placement group."
        )
        cluster_resources = params.pg_config.resource
        cluster_cpus = cluster_resources["CPU"]
        cluster_memory = cluster_resources["memory"]
        task_max_parallelism = cluster_cpus
        compaction_audit.set_total_cluster_memory_bytes(cluster_memory)
`

Reply:
The danger is that we will may use more the number of CPUs as parallelism which will fail the jobs.